### PR TITLE
GangaGUI - Jobs API

### PR DESCRIPTION
**Pull request by varunbankar**
### Summary

**Background**
`Jobs` API can be used to get the list of jobs (along with their general information) from the jobs repository. It can also be used to just get a list of ids of the jobs or a list of ids of the incomplete jobs.

**Work Overview (3 Endpoints | 3 Tests)**
Following endpoints are created for the `Jobs` API:
* `/api/jobs` - `GET` - Get a list of jobs (along with their general information) in the jobs repository.
* `/api/jobs/ids` - `GET` - Get the list of ids of the jobs in the jobs repository.
* `/api/jobs/incomplete_ids` - `GET` - Get the list of ids of the jobs that are incomplete in the jobs repository.

---

### How to Use the API

**Step 1: Start the GUI Server**

ganga_start_webserver

**Step 2: Generate the Token**

ganga_generate_token

**Step 3a: `/api/jobs` - `GET` Endpoint**

ganga_jobs

**Step 3b: `/api/jobs/ids` - `GET` Endpoint**

ganga_jobs_ids

**Step 3c: `/api/jobs/incomplete_ids` - `GET` Endpoint**

ganga_jobs_incomplete

---

__The start page for this project can be accessed [here](https://github.com/ganga-devs/ganga/issues/1663).__

### CODE REVIEW

File: ganga/GangaGUI/test/test_jobs_api.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# ******************** Imports ******************** #
+
```
Comment (egede) [inline ganga/GangaGUI/test/test_jobs_api.py]: Try to avoid self-evident comment lines.

**Review by egede — DISMISSED** [review]

**Comment by egede** [comment]
I am confused about the relationship between this PR and #1700 . Are they from the same branch?

File: ganga/GangaGUI/test/test_jobs_api.py (hunk 1 above)
Comment (varunbankar) [reply ganga/GangaGUI/test/test_jobs_api.py]: Okay, I will make the changes in the next PRs

**Comment by varunbankar** [comment]
> I am confused about the relationship between this PR and #1700 . Are they from the same branch?

This is a different branch. The `Jobs` API will be used in order to display the tabular form of the job repository in the frontend. Whereas `Job` API will be used to get more information about a single job.

**Review by egede — DISMISSED** [review]

**Review by egede — APPROVED** [review]

**Pull request merged**